crates/sui-types/src/proto_value.rs

```diff
@@ -73,30 +72,10 @@ impl ProtoVisitorBuilder {
         mut self,
         bytes: &[u8],
         layout: &A::MoveTypeLayout,
-    ) -> anyhow::Result<Value> {
+    ) -> Result<Value, Error> {
         let mut visitor = self.new_visitor()?;
-
         A::MoveValue::visit_deserialize(bytes, layout, &mut visitor)
     }
-
-    /// Deserialize `bytes` as a `MoveStruct` with layout `layout`. Can fail if the bytes do not
-    /// represent a struct with this layout, or if the deserialized struct exceeds the field/type
-    /// size budget.
-    pub fn deserialize_struct(
```

Comment:
@bmwill, I ended up removing this because I don't think it's currently used, and because of how we handle specific structs (strings, URLs, optionals), the "Expected to deserialize a struct" error can trigger erroneously.